### DISCUSSION TITLE
Is it possible to show current function name/signature?

### DISCUSSION THREAD

**Discussion by vitaly-zdanevich** [neoclide/coc.nvim#4194] [Q&A]
In TypeScript.

Current for the cursor position.

**Comment by chemzqm** [comment]
:h b:coc_current_function

**Reply by vitaly-zdanevich** [reply]
```
b:coc_current_function					*b:coc_current_function*

	Function string that current cursor in.

	Set `"coc.preferences.currentFunctionSymbolAutoUpdate": true`
	in coc-settings.json to update it on CursorHold.
```
And how to use it?

**Reply by fannheyward** [reply]
1. enable `coc.preferences.currentFunctionSymbolAutoUpdate`
2. you can add `b:coc_current_function` to your stausline functions.

**Reply by vitaly-zdanevich** [reply]
Can I show function name not on statusline but in that another line, where I see `:commands...`?

Or can I have a shortcut? For example in Vim we see file path by Ctrl-G.

**Reply by chemzqm** [reply]
You can use `:echo b:coc_current_function`

**Reply by vitaly-zdanevich** [reply]
Got `E121 Undefined variable`.

**Reply by vitaly-zdanevich** [reply]
This is some defect or my side?

**Reply by chemzqm** [reply]
Check the help by `:h b:coc_current_function`
